examples/tree-view/src/Node/Node.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import {
+  addChild,
+  createNode,
+  deleteNode,
+  removeChild,
+  increment
+} from './treeSlice'
+
+const Node = React.memo(({ id, parentId }) => {
+  const node = useSelector(state => state.tree[id])
+  const dispatch = useDispatch()
+
+  const { childIds, counter } = node
+
+  const handleIncrementClick = () => {
+    dispatch(increment(id))
+  }
+
+  const handleAddChildClick = () => {
+    dispatch(createNode())
+    dispatch(addChild(id))
+  }
+
+  const handleRemoveClick = () => {
+    dispatch(removeChild({ nodeId: parentId, childId: id }))
+    dispatch(deleteNode(id))
+  }
+
+  const renderChild = ({ childId, id }) => {
```

Comment:
:hand: Does this really need to be defined as a separate function?  I'd say it's more idiomatic to have an inline callback for `map()`